internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -287,6 +292,11 @@ func (a *Adapter) EnsureIntegrationPipelineRunsExist() (controller.OperationResu
 					if !clienterrors.IsInvalid(err) {
 						errsForPLRCreation = errors.Join(errsForPLRCreation, err)
 					}
+					err = a.checkAndCancelOldSnapshotsPipelineRun(a.application, a.snapshot)
```

Comment:
If we're cancelling all integration pipelines for a given Snapshot, I think this line should be called after we've ensured that the new pipelineRuns are all running - so not within this `for` loop, but after it.

Reply:
Good point, moved this part after the loop.